modules/setting/setting.go

```diff
@@ -1165,7 +1168,11 @@ func newService() {
 	Service.NoReplyAddress = sec.Key("NO_REPLY_ADDRESS").MustString("noreply.example.org")
 
 	sec = Cfg.Section("openid")
-	Service.EnableOpenIDSignIn = sec.Key("ENABLE_OPENID_SIGNIN").MustBool(false)
+	defOpenIDSignIn := false;
+	if customConfExists {
+		defOpenIDSignIn = true;
+	}
+	Service.EnableOpenIDSignIn = sec.Key("ENABLE_OPENID_SIGNIN").MustBool(defOpenIDSignIn)
 	Service.EnableOpenIDSignUp = sec.Key("ENABLE_OPENID_SIGNUP").MustBool(!Service.DisableRegistration && Service.EnableOpenIDSignIn)
```

Comment:
I think this should simply be
```
Service.EnableOpenIDSignIn = sec.Key("ENABLE_OPENID_SIGNIN").MustBool(!InstallLock)
```
because `InstallLock` state seems to indicate whether this is fresh install or not, while Gitea itself creates `app.ini` during install process.